app/src/main/java/com/alfresco/content/app/activity/ExtensionActivity.kt

```diff
@@ -0,0 +1,69 @@
+package com.alfresco.content.app.activity
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.appcompat.app.AppCompatActivity
+import androidx.navigation.findNavController
+import com.airbnb.mvrx.MavericksView
+import com.airbnb.mvrx.withState
+import com.alfresco.content.activityViewModel
+import com.alfresco.content.app.R
+import com.alfresco.content.app.widget.ActionBarController
+
+/**
+ * Marked as ExtensionActivity class
+ */
+class ExtensionActivity : AppCompatActivity(), MavericksView {
+
+    private val viewModel: MainActivityViewModel by activityViewModel()
+    private val navController by lazy { findNavController(R.id.nav_host_fragment) }
+    private lateinit var actionBarController: ActionBarController
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_extension)
+
+        when (intent?.action) {
+            Intent.ACTION_SEND -> {
+                println("EXTENSION SINGLE IMAGE")
+                if (intent.type?.startsWith("image/") == true) {
```

Comment:
It should handle multiple types of files like images, video, pdf etc?

Reply:
I've not worked on that part yet so will do it later along with the planned story.